_source/_posts/2020-12-09-dotnet-cloud-host-publish.md

```diff
@@ -0,0 +1,325 @@
+
+{% img blog/dotnetcore-deploy-major-cloud-hosts/image11.png alt:"" width:"800" %}{: .center-image }
+
+On the General tab hit Edit by the General Settings section:
+
+{% img blog/dotnetcore-deploy-major-cloud-hosts/image12.png alt:"" width:"800" %}{: .center-image }
+
+Set the following URLs:
+*Login redirect URIs*: `http://<your_app_url>/authorization-code/callback`
+e.g.: `http://oktaexample-env.eba-evyiprzp.us-east-1.elasticbeanstalk.com/authorization-code/callback`
+
+*Logout redirect URIs*: `http://<your_app_url>/signout/callback`
+e.g.: `http://oktaexample-env.eba-evyiprzp.us-east-1.elasticbeanstalk.com/signout/callback`
+
+Now you should be able to log in and log out in your web application using your Okta credentials.
+
+### Troubleshooting
+
+If you run into a Correlation Error after logging in, you'll need to manually set the `SameSite` cookie attribute to `None`, and enable SSL (HTTPS) on your server. I have a detailed article and a guide coming soon on how to set up a Linux server to support Okta login in .NET Core 3.1 applications - be on the lookout for that! To get an idea of the basics, check out my post on how to [Install .NET Core Apps on Linux in 5 Minutes](https://developer.okta.com/blog/2020/11/25/how-to-install-dotnetcore-on-linux).
+
+## Deploying a .NET Core 3.1 App to Google App Engine
+
+### Preparing Google Cloud App Engine
+
+Open the [Google Cloud Platform Console](https://console.cloud.google.com) and create a new Project:
+
+{% img blog/dotnetcore-deploy-major-cloud-hosts/image13.png alt:"" width:"600" %}{: .center-image }
+
+On the welcome screen select *Create Application*.
+
+{% img blog/dotnetcore-deploy-major-cloud-hosts/image14.png alt:"" width:"500" %}{: .center-image }
+
+Choose any Region. Then add *.NET* as Language and *Flexible* as Environment.
+
+{% img blog/dotnetcore-deploy-major-cloud-hosts/image15.png alt:"" width:"700" %}{: .center-image }
+
+Open a terminal or  command prompt, go to the project folder and run:
+
+```sh
+dotnet publish -o my-app
+```
+
+Then go to the newly created folder:
+
+```sh
+cd my-app
+```
+
+Log in with the Cloud SDK:
+
+```sh
+gcloud init
+```
+
+### Deploying the App to AWS Beanstalk
+
+To deploy the application enter:
+
+```sh
+gcloud app deploy
+```
+
+You might be asked a few questions while the deployment is running.
+
+Once the deployment is finished you can go back to the Cloud Platform Console and click the service to see if the application is running:
+
+{% img blog/dotnetcore-deploy-major-cloud-hosts/image16.png alt:"" width:"700" %}{: .center-image }
+
+### Configuring Okta for the Google Cloud Hosted App
+
+We need to add our application's URL to Okta to be recognized as a redirect URL.
+
+Going back to the Okta Dashboard Applications page, select the App you created in the first part.
+
+{% img blog/dotnetcore-deploy-major-cloud-hosts/image17.png alt:"" width:"800" %}{: .center-image }
```

Comment:
This image is blown up bigger than the original dimensions and looks fuzzy. Maybe create a new version?

From the preview: https://d33wubrfki0l68.cloudfront.net/e0339271d25c60fd8a5cc281e6788c898cd58f03/a6d04/assets-jekyll/blog/dotnetcore-deploy-major-cloud-hosts/image17-0fb40531e6439da98b6a6e60ff8b899203ceb42e214d986e251cfa68d9ace529.png

Reply:
It looks fine inside of the Google Doc. Not sure about why that is.

Reply:
Resized